core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -2584,6 +2584,19 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
       }
     }
 
+  /** Outputs a singleton pure stream if the stream is empty.
+   *
+   * @example {{{
+   * scala> Stream.empty.unconsNonEmpty(0).toList
+   * res0: List[Int] = List(0)
+   * }}}
+   */
+  def unconsNonEmpty[O2 >: O](o: => O2): Stream[F, O2] =
+    this.pull.uncons1.flatMap {
+      case Some((hd, tl)) => Pull.output1(hd) >> tl.underlying
+      case None           => Pull.output1(o) >> Pull.done
+    }.stream
```

Comment:
So, using `uncons1` involves extracting and partitioning chunks. We can just use `uncons` for this: 
```suggestion
    this.pull.uncons.flatMap {
      case Some((hd, tl)) => Pull.output(hd) >> tl.underlying
      case None           => Pull.output1(o) >> Pull.done
    }.stream
```
Note that `Pull` is guaranteed to only contain non-empty chunks.

Reply:
@diesalbla good point, thanks. Done.